src/query/procedure/mg_procedure_impl.cpp

```diff
@@ -2250,6 +2250,39 @@ mgp_error mgp_graph_delete_edge(struct mgp_graph *graph, mgp_edge *edge) {
   });
 }
 
+namespace {
+void NextPermitted(mgp_vertices_iterator &it) {
+  const auto *checker = it.graph->ctx->fine_grained_access_checker;
+
+  if (!checker) {
+    return;
+  }
+
+  while (it.current_it != it.vertices.end()) {
+    auto labels = (*it.current_it).impl_.Labels(it.graph->view);
+    if (!labels.HasValue()) {
+      break;
+    }
```

Comment:
What does it mean that we cannot access the labels of the vertex? It means we don't have access to the vertex at that point. So instead of breaking, I think we should just move to the next vertex. I think the iterator takes care of this, so this should be an assert, like `MG_ASSERT(labels.HasValue(), "Unexpected error while getting the labels of a vertex!")`, or something, the error message is just the first that popped into my mind.

```suggestion
    MG_ASSERT(labels.HasValue(), "Unexpected error while getting the labels of a vertex!");
```

Reply:
I will fix this in the next PR, if You are OK with that